core/eolearn/tests/test_core_tasks.py

```diff
@@ -71,38 +72,33 @@ def patch_fixture():
     return patch
 
 
-def test_copy(patch):
-    patch_copy = CopyTask().execute(patch)
-
-    assert patch == patch_copy, "Copied patch is different"
-
-    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
+@pytest.mark.parametrize("task", [DeepCopyTask, CopyTask])
+def test_copy(task: CopyTask, patch: EOPatch) -> None:
+    patch_copy = task().execute(patch)
+    assert patch_copy == patch
 
     patch_copy.data["bands"][0, 0, 0, 0] += 1
-    assert np.array_equal(patch.data["bands"], patch_copy.data["bands"]), "Data should not be copied"
-
-
-def test_deepcopy(patch):
-    patch_deepcopy = DeepCopyTask().execute(patch)
-
-    assert patch == patch_deepcopy, "Deep copied patch is different"
+    assert (patch_copy != patch) if task == DeepCopyTask else (patch_copy == patch)
 
-    patch_deepcopy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
+    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
+    assert "new" not in patch.data
 
-    patch_deepcopy.data["bands"][0, 0, 0, 0] += 1
-    assert not np.array_equal(patch.data["bands"], patch_deepcopy.data["bands"]), "Data should be copied"
 
+@pytest.fixture(name="expected_patch")
+def expected_patch_fixture(patch, request) -> EOPatch:
+    return EOPatch(**{feature: getattr(patch, feature) for feature in request.param})
 
-def test_partial_copy(patch):
-    partial_copy = DeepCopyTask(features=[(FeatureType.MASK_TIMELESS, "mask"), FeatureType.BBOX]).execute(patch)
-    expected_patch = EOPatch(mask_timeless=patch.mask_timeless, bbox=patch.bbox)
-    assert partial_copy == expected_patch, "Partial copying was not successful"
 
-    partial_deepcopy = DeepCopyTask(features=[FeatureType.TIMESTAMP, (FeatureType.SCALAR, "values")]).execute(patch)
-    expected_patch = EOPatch(scalar=patch.scalar, timestamp=patch.timestamp)
-    assert partial_deepcopy == expected_patch, "Partial deep copying was not successful"
+@pytest.mark.parametrize(
+    "features, expected_patch",
+    [
+        ([(FeatureType.MASK_TIMELESS, "mask"), FeatureType.BBOX], ["mask_timeless", "bbox"]),
+        ([FeatureType.TIMESTAMP, (FeatureType.SCALAR, "values")], ["scalar", "timestamp"]),
+    ],
+    indirect=["expected_patch"],
+)
+def test_partial_copy(features: List[FeatureSpec], patch: EOPatch, expected_patch: EOPatch) -> None:
+    assert DeepCopyTask(features=features).execute(patch) == expected_patch
```

Comment:
Ah, I see I was too hasty, the features need to be taken from the `patch` fixture. Then I guess we just need to change the line where `patch_expected` is defined into:

```python
patch_expected = EOPatch(**{feature_type: patch[feature] for feature_type, feature in features_dict.items()})
```

Reply:
Code was updated is this still relevant ?

Reply:
@zigaLuksic how do you feel about the suggestion in this thread? In order to compare all the features a new EOPatch can be created and the eopatch comparison is used, instead of looping over features and doing basically the same thing as the eopatch comparison does behind the scenes. The latter is currently implemented, but I would suggest the former since it looks cleaner, but I'm not aware of any potential performance drops or anything in that direction.

Reply:
after talking with @mlubej we decided that this is a better approach (as it does not use an ad-hoc implementation).

However it can be simplified. No reason to iterate and compare features. The features can be extracted with `get_features()` and compared as subsets. Then all existing features are iterated over and checked that they match (as is done now).

Reply:
done